### ISSUE TITLE
Multiple Split Horizon DNS

### ISSUE THREAD

**Issue by ishworg** [tailscale/tailscale#5530]
G'day Tailscale team and especially my hello to @DentonGentry from down under! :) 

# TL;DR question

Is multiple split horizon DNS supported yet in Tailscale? 

# Situation

We intend to add multiple Route53 hosted zone in future. Our infra delivery pipeline deploys things to its own environment.

On tailscale console:

Split DNS set for:
`foo.bar.internal` -> 10.247.0.2 (aws r53 HZ account A)
`wilma.bar.internal` -> 10.248.0.2 (aws r53 HZ account B)

Default DNS server:
10.247.0.2
10.248.0.2
PublicResolverDNSA
PublicResolverDNSB

The setting we have tested does not seem to show it works yet but there might be a missing knob somewhere (sent email + support tickets but lost in ether eh? :) 
- Client is on Manjaro `5.19.1-3`-MANJARO with tailscale and tailscaled on `1.28.0`
- Tailscaled on the server is on Ubuntu 20.04 with tailscale and tailscaled on `1.26.0`
- `10.247.0.0/24` is an exit node, `10.248.0.0/24` is *NOT* an exit node.

On Manjaro client /etc/resolv.conf is:
```
cat /etc/resolv.conf
# resolv.conf(5) file generated by tailscale
# DO NOT EDIT THIS FILE BY HAND -- CHANGES WILL BE OVERWRITTEN

nameserver 100.100.100.100
search foo.bar.internal wilma.bar.internal
```

Resolvage from the desktop client on Linux for `service.wilma.bar.internal` shows:
```
$ dig service.wilma.bar.internal

; <<>> DiG 9.18.5 <<>> service.wilma.bar.internal
;; global options: +cmd
;; connection timed out; no servers could be reached
```
At the same time, tailscale journal log on desktop client on Manjaro shows:
```
dns udp query: 500 Internal Server Error
```
Further, the DNS packet never reaches `10.248.0.2` by way of the tailscale router for `10.248.0.0/24`.

A successful resolvage from the desktop client on Linux for `service.foo.bar.internal` shows:
```
$ dig service.foo.bar.internal

; <<>> DiG 9.18.5 <<>> service.foo.bar.internal
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 31103
;; flags: qr rd ra; QUERY: 1, ANSWER: 1, AUTHORITY: 0, ADDITIONAL: 1

;; OPT PSEUDOSECTION:
; EDNS: version: 0, flags:; udp: 4095
;; QUESTION SECTION:
;service.foo.bar.internal. IN A

;; ANSWER SECTION:
service.foo.bar.internal. 60 IN A 10.247.0.19

;; Query time: 9 msec
;; SERVER: 100.100.100.100#53(100.100.100.100) (UDP)
;; WHEN: Fri Sep 02 12:16:20 AEST 2022
;; MSG SIZE  rcvd: 94
```

ts netcheck from client:
```
Report:
	* UDP: true
	* IPv4: yes, 1.2.3.4:55990
	* IPv6: no
	* MappingVariesByDestIP: false
	* HairPinning: false
	* PortMapping: UPnP
	* Nearest DERP: Sydney
	* DERP latency:
		- syd: 14.5ms  (Sydney)
		- sin: 103.3ms (Singapore)
		- tok: 110.3ms (Tokyo)
		- blr: 139.9ms (Bangalore)
		- sfo: 160.8ms (San Francisco)
		- dfw: 192.9ms (Dallas)
		- ord: 206.9ms (Chicago)
		- sea: 227.3ms (Seattle)
		- nyc: 229.1ms (New York City)
		- lhr: 267.4ms (London)
		- fra: 301.3ms (Frankfurt)
		- sao:         (São Paulo)
```

At this point, I am not entirely sure where the issue is. It _appears_ the issue is from Tailscale DNS handling code but I could be wrong.

Thanks! :)

**Comment by DentonGentry** [comment]
For anyone else following along:
- Multiple SplitDNS domains do work. Prior to Tailscale 1.25.98, up to 50 Split Domains can be configured before the Windows client hits a limit in how it programs NRPT entries. The NRPT issue was fixed by [PR4742](https://github.com/tailscale/tailscale/pull/4742). In Tailscale 1.26.0 and later there is not a known limit in the number of SplitDNS domains one can have.
- We are corresponding on a support@tailscale.com about the particulars of this case.

**Comment by kim0** [comment]
fwiw, I'm using split DNS as well and just hit `dns udp query: 500 Internal Server Error` using tailscale on a macos client (v1.30.0). VPN DNS resolving is non functional.

**Comment by DentonGentry** [comment]
Please open a new issue, with details. You are commenting on a number of closed DNS-related issues, and it is highly unlikely that you are hitting the same root cause.
